File: 01_Greedy/06_모험가길드.py
n = int(input())

data = list(map(int, input().split()))
data.sort()

result = 0 
cnt = 0 # 현재 그룹에 포함된 모험가의 수

for i in data :
    cnt += 1
    if cnt >= i : # 현재 그룹에 포함된 모험가의 수가 현재의 공포도 이상이라면, 그룹 결성
        result += 1
        cnt = 0

print(result)

File: 01_Greedy/06_모험가길드.cpp
#include <iostream>
#include <vector>

using namespace std;

int main(void){

    int result = 0;
    int traveler;
    cin >> traveler;

    vector <int> scary;

    int user_input;
    for (int i = 0; i < traveler; i++)
    {
        cin >> user_input;
        scary.push_back(user_input);
    }
    sort(scary.begin(), scary.end());

    int cnt = 0;
    for (int i = 0; i < traveler; i++)
    {
        cnt += 1;
        if(cnt >= scary[i]) {
            result += 1;
            cnt = 0;
        }
    }
    
    cout << result << endl;

    return 0;
}

/*

5
2 3 1 2 2

*/